### PR TITLE
Add new server:openapi route

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## What does this PR do ?

Add new route `server:openapi` to the server controller

### How should this be manually tested?

* Start a Kuzzle stack
* test the two URL `http://localhost:7512/_swagger&format=json` and `http://localhost:7512/_swagger&format=yaml`

### Boyscout 

* Clean JS import in serverController
* Lint using new rules

### CODE REVIEW

File: lib/api/controller/server.js [hunk 1]
```diff
@@ -235,6 +237,28 @@ class ServerController extends NativeController {
     return apiDefinition;
   }
 
+  async swagger (request) {
+    const format = 
+      // The YAML response format is only valid with an HTTP request
+      request.context.connection.protocol === 'http'&& request.input.args.format
```
Comment (Aschen) [inline lib/api/controller/server.js:243]: Why can't we have YAML in WebSocket or MQTT? YAML it's just a string so we could put it in the result

File: package.json [hunk 2]
```diff
@@ -96,6 +96,7 @@
     "codecov": "^3.7.2",
     "cucumber": "^6.0.5",
     "eslint": "^7.6.0",
+    "js-yaml": "^3.14.0",
```
Comment (Aschen) [inline package.json:99]: Did you remove the former package `jsonToYaml` ?

**Review by Aschen — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (alexandrebouthinon) [reply package.json:99]: Nop, `jsonToYaml` is only for JSON to YAML conversion but I need YAML to JSON conversion in unit tests to performs a few checks (this is why I add it in the `devDependencies` only)

**Review by Aschen — APPROVED** [review]

**Review by Aschen — CHANGES_REQUESTED** [review]
IMHO we should change the `.kuzzlerc.sample` and also the `config.default.js` files to give the right to the anonymous user to call this action

**Review by Aschen — APPROVED** [review]

**Review by scottinet — APPROVED** [review]

**Pull request merged**